services/thermal_mgr/thermal_mgr.c

```diff
@@ -24,48 +24,93 @@ static uint8_t thermalMgrQueueStorageArea[THERMAL_MGR_QUEUE_LENGTH * THERMAL_MGR
 
 static void thermalMgr(void *pvParameters);
 
-void initThermalSystemManager(lm75bd_config_t *config) {
+void initThermalSystemManager(lm75bd_config_t *config)
+{
   memset(&thermalMgrTaskBuffer, 0, sizeof(thermalMgrTaskBuffer));
   memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
-  
+
   thermalMgrTaskHandle = xTaskCreateStatic(
-    thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
-    config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
+      thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
+      config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
 
   memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
   memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
 
   thermalMgrQueueHandle = xQueueCreateStatic(
-    THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
-    thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
-
+      THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
+      thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
 }
 
-error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event)
+{
+  // Check parameter validity
+  if (event == NULL)
+    return ERR_CODE_INVALID_ARG;
 
+  if (xQueueSend(thermalMgrQueueHandle, event, 0) != pdTRUE)
+  {
+    return ERR_CODE_QUEUE_FULL;
+  }
   return ERR_CODE_SUCCESS;
 }
 
-void osHandlerLM75BD(void) {
-  /* Implement this function */
+void osHandlerLM75BD(void)
+{
+  thermal_mgr_event_t interrupt;
+  interrupt.type = THERMAL_MGR_EVENT_OS_INTERRUPT;
+  thermalMgrSendEvent(&interrupt);
 }
 
-static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
-  while (1) {
-    
+static void thermalMgr(void *pvParameters)
+{
+  lm75bd_config_t *config = (lm75bd_config_t *)pvParameters;
+  thermal_mgr_event_t receivedEvent;
+  while (1)
+  {
+    if (xQueueReceive(thermalMgrQueueHandle, &receivedEvent, portMAX_DELAY) == pdTRUE)
```

Comment:
Limit scope of receivedEvent
```suggestion
  while (1)
  {
      thermal_mgr_event_t receivedEvent = {0};
      if (xQueueReceive(thermalMgrQueueHandle, &receivedEvent, portMAX_DELAY) == pdTRUE)
```